### PR TITLE
Fix Material 3 tab indicator weight and position

### LINKED ISSUE #123112: TabBar height and vertical text alignment don't match spec
## Steps to Reproduce

1. Execute `flutter run` on the code sample. 
2. Observe the TabBar height.
3. Observe the vertical text alignment.

**Expected results:** 
Height and vertical text alignment match the [M3 specs](https://m3.material.io/components/tabs/specs#471acc02-600f-4fe2-8651-70549d4a6679).

**Actual results:** 
Height and vertical text alignment don't match the M3 specs.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    final theme = ThemeData.from(
      colorScheme: const ColorScheme.light(),
      textTheme: Typography.englishLike2021,
      useMaterial3: true,
    );

    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Text(
                'Expected results',
                style: theme.textTheme.titleMedium?.copyWith(
                  color: theme.colorScheme.onInverseSurface,
                ),
                textAlign: TextAlign.center,
              ),
              Text(
                'Height: 48.0\n'
                'Text vertically centered',
                style: theme.textTheme.bodyMedium?.copyWith(
                  color: theme.colorScheme.onInverseSurface,
                ),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 16.0),
              DefaultTabController(
                length: 3,
                child: Container(
                  color: theme.colorScheme.surface,
                  child: ConstrainedBox(
                    constraints: const BoxConstraints(maxHeight: 47.5),
                    child: const TabBar(
                      tabs: [
                        Tab(
                          child: Padding(
                            padding: EdgeInsetsDirectional.only(top: 2.5),
                            child: Text('Tab 1'),
                          ),
                        ),
                        Tab(
                          child: Padding(
                            padding: EdgeInsetsDirectional.only(top: 2.5),
                            child: Text('Tab 2'),
                          ),
                        ),
                        Tab(
                          child: Padding(
                            padding: EdgeInsetsDirectional.only(top: 2.5),
                            child: Text('Tab 3'),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
              const SizedBox(height: 32.0),
              Text(
                'Actual results',
                style: theme.textTheme.titleMedium?.copyWith(
                  color: theme.colorScheme.onInverseSurface,
                ),
                textAlign: TextAlign.center,
              ),
              Text(
                'Height: 48.67\n'
                'Text not vertically centered',
                style: theme.textTheme.bodyMedium?.copyWith(
                  color: theme.colorScheme.onInverseSurface,
                ),
                textAlign: TextAlign.center,
              ),
              const SizedBox(height: 16.0),
              DefaultTabController(
                length: 3,
                child: Container(
                  color: theme.colorScheme.surface,
                  child: const TabBar(
                    tabs: [
                      Tab(text: 'Tab 1'),
                      Tab(text: 'Tab 2'),
                      Tab(text: 'Tab 3'),
                    ],
                  ),
                ),
              ),
            ],
          ),
        ),
        backgroundColor: theme.colorScheme.inverseSurface,
      ),
      theme: theme,
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel stable, 3.7.7, on macOS 13.2.1 22D68 darwin-x64, locale en-US)
    • Flutter version 3.7.7 on channel stable at /Users/anthonymoretti/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ad6cd72c0 (13 days ago), 2023-03-08 09:41:59 -0800
    • Engine revision 1837b5be5f
    • Dart version 2.19.4
    • DevTools version 2.20.1

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed
      instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed
      instructions).

[✓] VS Code (version 1.76.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.60.0

[✓] Connected device (3 available)
    • iPhone 14 Pro (mobile) • ED506CAA-EB76-47D8-BBDC-751CB0EE7EC8 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 13.2.1
      22D68 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome
      111.0.5563.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

Tab issue

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/123112

### Description
1. Add proper M3 indicator height aka`IndictorWeight` from the M3 specs for the primary tab bar with label indicator size.
https://github.com/flutter/flutter/blob/db6074ade4e4fde664e6258d671faf356e1b6e85/dev/tools/gen_defaults/data/navigation_tab_primary.json#L9
(this was held due to `indicatorWeight` having a hard-coded value) 

and added a secondary tab bar indicator height.
2. Set a minimum value for the rounded indicator to maintain the indicator shape.
3. With proper indicator height, the rounded indicator position is also fixed.
4. Fix round indicator is shown for the primary tab bar with tab indicator size.
5. Above changes fix https://github.com/flutter/flutter/issues/123112.
6. Fix the `startOffset` const value from https://github.com/flutter/flutter/pull/125036  to match docs and move it to a variable.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -821,8 +822,16 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
 
   /// The thickness of the line that appears below the selected tab.
   ///
-  /// The value of this parameter must be greater than zero and its default
-  /// value is 2.0.
+  /// The value of this parameter must be greater than zero.
+  ///
+  /// If [ThemeData.useMaterial3] is true and [TabBar] is a primary type,
+  /// the default value is 3.0. If the provided value is less than 3.0, the
```
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:828]: Why is it overridden for values < 3.0?

File: packages/flutter/lib/src/material/tabs.dart [hunk 2]
```diff
@@ -821,8 +822,16 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
 
   /// The thickness of the line that appears below the selected tab.
   ///
-  /// The value of this parameter must be greater than zero and its default
-  /// value is 2.0.
+  /// The value of this parameter must be greater than zero.
+  ///
+  /// If [ThemeData.useMaterial3] is true and [TabBar] is a primary type,
```
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:827]: Can this section mention what is a primary TabBar and what is secondary (the secondary constructor)? If folks don't know, they may not be able to understand what this means.

File: packages/flutter/lib/src/material/tabs.dart [hunk 3]
```diff
@@ -1186,19 +1195,25 @@ class _TabBarState extends State<TabBar> {
       color = Colors.white;
     }
 
+    final bool primaryWithLabelIndicator = widget._isPrimary && indicatorSize == TabBarIndicatorSize.label;
+    final double effectiveIndicatorWeight = theme.useMaterial3
```
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:1199]: This looks like if `useMaterial3` is true, the user provided value of widget.indicatorWeight is just ignored completely?

Comment (TahaTesser) [reply packages/flutter/lib/src/material/tabs.dart:1199]: Good catch!

File: packages/flutter/lib/src/material/tabs.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/tabs.dart:828]: This preserves the rounded indicator shape and overrides the `2.0` indicator weight. 

https://github.com/flutter/flutter/blob/4d38f94e27d4c7691d8f4eb6cc9b068db666642d/packages/flutter/lib/src/material/tabs.dart#L715
Screenshot 2023-05-03 at 14 17 00


If the value is less than `3.0`, this indicator shape cannot be maintained. This has rounded corners, unlike the tab indicator size indicator.

Screenshot 2023-05-03 at 14 14 00

File: packages/flutter/lib/src/material/tabs.dart (hunk 2 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/tabs.dart:827]: Great idea

**Review by Piinks — APPROVED** [review]
LGTM!

**Comment by Piinks** [comment]
Goldens are approved. Thank you!

**Pull request merged**